File: CHEFSTLT.py
t = input()
while(t > 0):
	t -= 1
	s1 = raw_input()
	s2 = raw_input()
	qb=0;qs=0;same=0;dif=0;
	for i in range (0,len(s1)):

		if(s1[i] == s2[i]):
			same += 1
			if(s1[i] == '?'):
				qb += 1
		
		if(s1[i] != s2[i]):
			dif += 1
			if( (s1[i] == '?') | (s2[i] == '?') ):
				qs += 1

	print len(s1)-same-qs,
	print dif+qb

File: chef/CHEFSTLT.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
	int t;
	cin >> t;
	while(t--)
	{
		string s1,s2;
		cin >> s1 >> s2;
		int i,qb=0,qs=0,same=0,dif=0;

		for(i=0; i<s1.length(); i++)
		{
			if(s1[i] == s2[i])
			{
				same++;
				if(s1[i] == '?')
					qb++;
			}
			if(s1[i] != s2[i])
			{
				dif++;
				if( (s1[i] == '?') || (s2[i] == '?') )
					qs++;
			}
		}
		cout << s1.length()-same-qs << " " << dif+qb << "\n";
	}
	return 0;
}